### PR TITLE
test: add unit test to verify command line flag generation for seccomp

### PR DESCRIPTION
**Pull request by bravebeaver**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: pkg/agent/utils_test.go [hunk 1]
```diff
@@ -743,6 +746,39 @@ var _ = Describe("Test GetOrderedKubeletConfigFlagString", func() {
...
+
+		ap := &datamodel.AgentPoolProfile{
+			CustomKubeletConfig: &datamodel.CustomKubeletConfig{
+				SeccompDefault: to.BoolPtr(false),
+			},
+		}
+
+		expectedStr := "--node-labels=topology.kubernetes.io/region=southcentralus --seccomp-default=true "
+		actualStr := GetOrderedKubeletConfigFlagString(k, cs, ap, true)
+		Expect(expectedStr).To(Equal(actualStr), fmt.Sprintf("Expected: %s\n, but got: %s", expectedStr, actualStr))
```
Comment (AlisonB319) [inline pkg/agent/utils_test.go:780]: make sure to import fmt if you want to use it here

File: pkg/agent/utils_test.go [hunk 2]
```diff
@@ -743,6 +746,39 @@ var _ = Describe("Test GetOrderedKubeletConfigFlagString", func() {
...
+		}
+
+		ap := &datamodel.AgentPoolProfile{
+			CustomKubeletConfig: &datamodel.CustomKubeletConfig{
+				SeccompDefault: to.BoolPtr(false),
+			},
+		}
+
+		expectedStr := "--node-labels=topology.kubernetes.io/region=southcentralus --seccomp-default=true "
+		actualStr := GetOrderedKubeletConfigFlagString(k, cs, ap, true)
```
Comment (AlisonB319) [inline pkg/agent/utils_test.go:779]: Seems this function got updated to take a datamodel, I'd just do something like 
```
dm := datamodel.NodeBootstrappingConfiguration{
			KubeletConfig:    k,
			ContainerService: cs,
			AgentPoolProfile: ap,
		}
actualStr := GetOrderedKubeletConfigFlagString(&dm)
```

File: pkg/agent/utils_test.go (hunk 1 above)
Comment (cameronmeissner) [reply pkg/agent/utils_test.go:780]: yeah the extra message here is unecessary

**Comment by cameronmeissner** [comment]
do we already have e2e tests/validation that cover this scenario? if not might be worth adding as well

File: pkg/agent/utils_test.go (hunk 1 above)
Comment (bravebeaver) [reply pkg/agent/utils_test.go]: weirdly when i ran for the first time it did not print out on my terminal. rebase error 😅. updated

File: pkg/agent/utils_test.go (hunk 2 above)
Comment (bravebeaver) [reply pkg/agent/utils_test.go]: done.. rebased and resolved merge conflicts

File: pkg/agent/utils_test.go [hunk 3]
```diff
@@ -743,6 +746,39 @@ var _ = Describe("Test GetOrderedKubeletConfigFlagString", func() {
...
+							Config: map[string]string{
+								"--seccomp-default": "true",
+							},
+						},
+					},
+				},
+			},
+		}
+		k := map[string]string{
+			"--node-labels": "topology.kubernetes.io/region=southcentralus",
```
[the remark this answers is not in the document]
Comment (bravebeaver) [reply pkg/agent/utils_test.go]: to check whether the seccompDefault flag is properly passed down / merged

**Comment by bravebeaver** [comment]
> do we already have e2e tests/validation that cover this scenario? if not might be worth adding as well

that's my plan to add e2e down the path, first to understand how the configs are generated along the way and fill in unit test gaps.

**Comment by r2k1** [comment]
> do we already have e2e tests/validation that cover this scenario? if not might be worth adding as well

While e2e can be handy to use. I think we should avoid putting stuff there. It's better to test things on other layers when possible.
e2e are hard to debug. And reliability drops exponentially with increasing number of tests.

**Review by r2k1 — APPROVED** [review]

File: pkg/agent/utils_test.go [hunk 4]
```diff
@@ -712,10 +714,11 @@ var _ = Describe("Test GetOrderedKubeletConfigFlagString", func() {
 			AgentPoolProfile:        &datamodel.AgentPoolProfile{},
 		}
 
-		expectStr := "--event-qps=0 --image-gc-high-threshold=85 --node-status-update-frequency=20s --streaming-connection-idle-timeout=4h0m0s "
+		expectStr := "--event-qps=0 --image-gc-high-threshold=85 --node-status-update-frequency=20s --seccomp-default=true --streaming-connection-idle-timeout=4h0m0s "
```
Comment (r2k1) [inline pkg/agent/utils_test.go:717]: do you need more than 1 test?

**Review by timmy-wright — APPROVED** [review]

**Pull request merged**